engines/python/src/main/java/ai/djl/python/engine/Connection.java

```diff
@@ -127,46 +127,26 @@ static String[] getPythonStartCmd(PyEnv pyEnv, Model model, int workerId, int po
         if (pyEnv.isMpiMode()) {
             String cudaDevices = getVisibleDevices(workerId, tensorParallelDegree);
             logger.info("Set CUDA_VISIBLE_DEVICES={}", cudaDevices);
-            String[] args = new String[36];
-            args[0] = "mpirun";
-            args[1] = "-N";
+            pyEnv.addEnv("CUDA_VISIBLE_DEVICES", cudaDevices);
```

Comment:
This doesn't get passed through to the process's environment.

I tried running it, and it failed with OOM because all `torchrun` workers launched a process that connected to the same GPU.

Right before OOM, on AWS g5.12xlarge instance:
```
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A   3603150      C   ...11-openjdk-amd64/bin/java      248MiB |
|    2   N/A  N/A   3603288      C   /usr/bin/python3                 5542MiB |
|    2   N/A  N/A   3603289      C   /usr/bin/python3                 5750MiB |
|    2   N/A  N/A   3603290      C   /usr/bin/python3                 5542MiB |
|    2   N/A  N/A   3603293      C   /usr/bin/python3                 5542MiB |
+-----------------------------------------------------------------------------+
```

Reply:
this is weird, if customer is doing the right thing

```
 getVisibleDevices(workerId, tensorParallelDegree)
```

should bring you the right VISIBLE DEVICES based on the TP

Reply:
I think `PyEnv.addEnv()` only pass env var to master process of `torchrun` but child process doesn't get it.

That's why `mpirun` use `-x` args to send env to child process.

Reply:
I added a `print` statement to the `DeepSpeedService.__init__`:

```
[INFO ] PyProcess - W-3416-gptj-stdout: YOOOOOO LOOK AT THIS! LOCAL_RANK: 0,  CUDA_VISIBLE_DEVICES: 3
[INFO ] PyProcess - W-3416-gptj-stdout: Python engine started.
[INFO ] PyProcess - W-3417-gptj-stdout: YOOOOOO LOOK AT THIS! LOCAL_RANK: 0,  CUDA_VISIBLE_DEVICES: 3
[INFO ] PyProcess - W-3417-gptj-stdout: Python engine started.
[INFO ] PyProcess - W-3419-gptj-stdout: YOOOOOO LOOK AT THIS! LOCAL_RANK: 0,  CUDA_VISIBLE_DEVICES: 3
[INFO ] PyProcess - W-3419-gptj-stdout: Python engine started.
[INFO ] PyProcess - W-3418-gptj-stdout: YOOOOOO LOOK AT THIS! LOCAL_RANK: 0,  CUDA_VISIBLE_DEVICES: 3
[INFO ] PyProcess - W-3418-gptj-stdout: Python engine started.
```


The print statement:
``` python
print(f"YOOOOOO LOOK AT THIS! LOCAL_RANK: {os.getenv('LOCAL_RANK')},  CUDA_VISIBLE_DEVICES: {os.getenv('CUDA_VISIBLE_DEVICES')}")
```

So yeah, something else seems off.

Reply:
Here's some extra logging, where I logged all of `os.environ` from all workers, in case that helps you debug:
```
[INFO ] PyProcess - W-3894-gptj-stdout: environ({'PATH': '/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin', 'NV_LIBCUSPARSE_VERSION': '11.7.5.86-1', 'PYTORCH_KERNEL_CACHE_PATH': '/tmp/.cache', 'HUGGINGFACE_HUB_CACHE': '/tmp/.cache/huggingface/hub', 'NCCL_VERSION': '2.16.2-1', 'NV_NVTX_VERSION': '11.8.86-1', 'NVIDIA_REQUIRE_CUDA': 'cuda>=11.8 brand=tesla,driver>=450,driver<451 brand=tesla,driver>=470,driver<471 brand=unknown,driver>=470,driver<471 brand=nvidia,driver>=470,driver<471 brand=nvidiartx,driver>=470,driver<471 brand=geforce,driver>=470,driver<471 brand=geforcertx,driver>=470,driver<471 brand=quadro,driver>=470,driver<471 brand=quadrortx,driver>=470,driver<471 brand=titan,driver>=470,driver<471 brand=titanrtx,driver>=470,driver<471 brand=tesla,driver>=510,driver<511 brand=unknown,driver>=510,driver<511 brand=nvidia,driver>=510,driver<511 brand=nvidiartx,driver>=510,driver<511 brand=geforce,driver>=510,driver<511 brand=geforcertx,driver>=510,driver<511 brand=quadro,driver>=510,driver<511 brand=quadrortx,driver>=510,driver<511 brand=titan,driver>=510,driver<511 brand=titanrtx,driver>=510,driver<511 brand=tesla,driver>=515,driver<516 brand=unknown,driver>=515,driver<516 brand=nvidia,driver>=515,driver<516 brand=nvidiartx,driver>=515,driver<516 brand=geforce,driver>=515,driver<516 brand=geforcertx,driver>=515,driver<516 brand=quadro,driver>=515,driver<516 brand=quadrortx,driver>=515,driver<516 brand=titan,driver>=515,driver<516 brand=titanrtx,driver>=515,driver<516', 'JAVA_OPTS': '-Xmx1g -Xms1g -XX:+ExitOnOutOfMemoryError -XX:-UseContainerSupport', 'NV_LIBCUBLAS_PACKAGE_NAME': 'libcublas-11-8', 'NVIDIA_PRODUCT_NAME': 'CUDA', 'MXNET_ENGINE_TYPE': 'NaiveEngine', 'NV_CUDNN_VERSION': '8.9.0.131', 'CUDA_VISIBLE_DEVICES': '2', 'LD_LIBRARY_PATH': '/usr/local/nvidia/lib:/usr/local/nvidia/lib64', 'NV_NVPROF_DEV_PACKAGE': 'cuda-nvprof-11-8=11.8.87-1', 'PWD': '/opt/djl', 'NV_LIBNPP_VERSION': '11.8.0.86-1', 'NV_LIBNCCL_PACKAGE': 'libnccl2=2.16.2-1+cuda11.8', 'NV_LIBNPP_DEV_VERSION': '11.8.0.86-1', 'TF_CPP_MIN_LOG_LEVEL': '1', 'PYTHONPATH': '/tmp/.djl.ai/python/0.24.0-SNAPSHOT:/opt/ml/model/gptj', 'NV_CUDA_CUDART_VERSION': '11.8.89-1', 'NV_CUDA_NSIGHT_COMPUTE_VERSION': '11.8.0-1', 'NV_LIBCUBLAS_DEV_VERSION': '11.11.3.6-1', 'OLDPWD': '/opt/djl/src/djl-serving/serving', 'NVIDIA_DRIVER_CAPABILITIES': 'compute,utility', 'OMP_NUM_THREADS': '1', 'LIBRARY_PATH': '/usr/local/cuda/lib64/stubs', 'PREDICT_TIMEOUT': '240', 'NV_CUDA_NSIGHT_COMPUTE_DEV_PACKAGE': 'cuda-nsight-compute-11-8=11.8.0-1', 'NV_CUDNN_PACKAGE_NAME': 'libcudnn8', 'LS_COLORS': 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:', 'USE_AICCL_BACKEND': 'true', 'NV_LIBNCCL_DEV_PACKAGE_NAME': 'libnccl-dev', 'SHLVL': '2', 'NV_LIBCUSPARSE_DEV_VERSION': '11.7.5.86-1', 'NV_LIBNCCL_DEV_PACKAGE': 'libnccl-dev=2.16.2-1+cuda11.8', 'NV_CUDA_LIB_VERSION': '11.8.0-1', 'JAVA_HOME': '/usr/lib/jvm/java-11-openjdk-amd64', 'NV_CUDA_CUDART_DEV_VERSION': '11.8.89-1', 'CUDA_VERSION': '11.8.0', 'NVIDIA_VISIBLE_DEVICES': 'all', 'NV_NVPROF_VERSION': '11.8.87-1', 'NV_CUDA_COMPAT_PACKAGE': 'cuda-compat-11-8', 'TERM': 'xterm', 'NV_LIBNCCL_PACKAGE_NAME': 'libnccl2', 'BITSANDBYTES_NOWELCOME': '1', 'NV_NVML_DEV_VERSION': '11.8.86-1', 'MODEL_LOADING_TIMEOUT': '1200', 'NV_LIBCUBLAS_PACKAGE': 'libcublas-11-8=11.11.3.6-1', 'TF_NUM_INTRAOP_THREADS': '1', 'NV_LIBCUBLAS_DEV_PACKAGE_NAME': 'libcublas-dev-11-8', '_': '/usr/bin/djl-serving', 'NV_CUDNN_PACKAGE_DEV': 'libcudnn8-dev=8.9.0.131-1+cuda11.8', 'NV_LIBNPP_DEV_PACKAGE': 'libnpp-dev-11-8=11.8.0.86-1', 'TRANSFORMERS_CACHE': '/tmp/.cache/huggingface/transformers', 'NV_LIBCUBLAS_DEV_PACKAGE': 'libcublas-dev-11-8=11.11.3.6-1', 'NVARCH': 'x86_64', 'DJL_CACHE_DIR': '/tmp/.djl.ai', 'MODEL_SERVER_HOME': '/opt/djl', 'NV_LIBNCCL_PACKAGE_VERSION': '2.16.2-1', 'NV_LIBNPP_PACKAGE': 'libnpp-11-8=11.8.0.86-1', 'HOSTNAME': '59b2955e7640', 'NV_CUDNN_PACKAGE': 'libcudnn8=8.9.0.131-1+cuda11.8', 'HOME': '/root', 'NV_LIBCUBLAS_VERSION': '11.11.3.6-1', 'NV_LIBNCCL_DEV_PACKAGE_VERSION': '2.16.2-1', 'LC_CTYPE': 'C.UTF-8', 'LOCAL_RANK': '0', 'RANK': '0', 'GROUP_RANK': '0', 'ROLE_RANK': '0', 'ROLE_NAME': 'default', 'LOCAL_WORLD_SIZE': '1', 'WORLD_SIZE': '1', 'GROUP_WORLD_SIZE': '1', 'ROLE_WORLD_SIZE': '1', 'MASTER_ADDR': '127.0.0.1', 'MASTER_PORT': '19001', 'TORCHELASTIC_RESTART_COUNT': '0', 'TORCHELASTIC_MAX_RESTARTS': '0', 'TORCHELASTIC_RUN_ID': 'none', 'TORCHELASTIC_USE_AGENT_STORE': 'True', 'NCCL_ASYNC_ERROR_HANDLING': '1', 'TORCHELASTIC_ERROR_FILE': '/tmp/torchelastic_jovpw2xu/none_66fgwsqy/attempt_0/0/error.json', 'CUDA_MODULE_LOADING': 'LAZY'})
[INFO ] PyProcess - W-3894-gptj-stdout: Python engine started.
[INFO ] PyProcess - W-3892-gptj-stdout: environ({'PATH': '/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin', 'NV_LIBCUSPARSE_VERSION': '11.7.5.86-1', 'PYTORCH_KERNEL_CACHE_PATH': '/tmp/.cache', 'HUGGINGFACE_HUB_CACHE': '/tmp/.cache/huggingface/hub', 'NCCL_VERSION': '2.16.2-1', 'NV_NVTX_VERSION': '11.8.86-1', 'NVIDIA_REQUIRE_CUDA': 'cuda>=11.8 brand=tesla,driver>=450,driver<451 brand=tesla,driver>=470,driver<471 brand=unknown,driver>=470,driver<471 brand=nvidia,driver>=470,driver<471 brand=nvidiartx,driver>=470,driver<471 brand=geforce,driver>=470,driver<471 brand=geforcertx,driver>=470,driver<471 brand=quadro,driver>=470,driver<471 brand=quadrortx,driver>=470,driver<471 brand=titan,driver>=470,driver<471 brand=titanrtx,driver>=470,driver<471 brand=tesla,driver>=510,driver<511 brand=unknown,driver>=510,driver<511 brand=nvidia,driver>=510,driver<511 brand=nvidiartx,driver>=510,driver<511 brand=geforce,driver>=510,driver<511 brand=geforcertx,driver>=510,driver<511 brand=quadro,driver>=510,driver<511 brand=quadrortx,driver>=510,driver<511 brand=titan,driver>=510,driver<511 brand=titanrtx,driver>=510,driver<511 brand=tesla,driver>=515,driver<516 brand=unknown,driver>=515,driver<516 brand=nvidia,driver>=515,driver<516 brand=nvidiartx,driver>=515,driver<516 brand=geforce,driver>=515,driver<516 brand=geforcertx,driver>=515,driver<516 brand=quadro,driver>=515,driver<516 brand=quadrortx,driver>=515,driver<516 brand=titan,driver>=515,driver<516 brand=titanrtx,driver>=515,driver<516', 'JAVA_OPTS': '-Xmx1g -Xms1g -XX:+ExitOnOutOfMemoryError -XX:-UseContainerSupport', 'NV_LIBCUBLAS_PACKAGE_NAME': 'libcublas-11-8', 'NVIDIA_PRODUCT_NAME': 'CUDA', 'MXNET_ENGINE_TYPE': 'NaiveEngine', 'NV_CUDNN_VERSION': '8.9.0.131', 'CUDA_VISIBLE_DEVICES': '2', 'LD_LIBRARY_PATH': '/usr/local/nvidia/lib:/usr/local/nvidia/lib64', 'NV_NVPROF_DEV_PACKAGE': 'cuda-nvprof-11-8=11.8.87-1', 'PWD': '/opt/djl', 'NV_LIBNPP_VERSION': '11.8.0.86-1', 'NV_LIBNCCL_PACKAGE': 'libnccl2=2.16.2-1+cuda11.8', 'NV_LIBNPP_DEV_VERSION': '11.8.0.86-1', 'TF_CPP_MIN_LOG_LEVEL': '1', 'PYTHONPATH': '/tmp/.djl.ai/python/0.24.0-SNAPSHOT:/opt/ml/model/gptj', 'NV_CUDA_CUDART_VERSION': '11.8.89-1', 'NV_CUDA_NSIGHT_COMPUTE_VERSION': '11.8.0-1', 'NV_LIBCUBLAS_DEV_VERSION': '11.11.3.6-1', 'OLDPWD': '/opt/djl/src/djl-serving/serving', 'NVIDIA_DRIVER_CAPABILITIES': 'compute,utility', 'OMP_NUM_THREADS': '1', 'LIBRARY_PATH': '/usr/local/cuda/lib64/stubs', 'PREDICT_TIMEOUT': '240', 'NV_CUDA_NSIGHT_COMPUTE_DEV_PACKAGE': 'cuda-nsight-compute-11-8=11.8.0-1', 'NV_CUDNN_PACKAGE_NAME': 'libcudnn8', 'LS_COLORS': 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:', 'USE_AICCL_BACKEND': 'true', 'NV_LIBNCCL_DEV_PACKAGE_NAME': 'libnccl-dev', 'SHLVL': '2', 'NV_LIBCUSPARSE_DEV_VERSION': '11.7.5.86-1', 'NV_LIBNCCL_DEV_PACKAGE': 'libnccl-dev=2.16.2-1+cuda11.8', 'NV_CUDA_LIB_VERSION': '11.8.0-1', 'JAVA_HOME': '/usr/lib/jvm/java-11-openjdk-amd64', 'NV_CUDA_CUDART_DEV_VERSION': '11.8.89-1', 'CUDA_VERSION': '11.8.0', 'NVIDIA_VISIBLE_DEVICES': 'all', 'NV_NVPROF_VERSION': '11.8.87-1', 'NV_CUDA_COMPAT_PACKAGE': 'cuda-compat-11-8', 'TERM': 'xterm', 'NV_LIBNCCL_PACKAGE_NAME': 'libnccl2', 'BITSANDBYTES_NOWELCOME': '1', 'NV_NVML_DEV_VERSION': '11.8.86-1', 'MODEL_LOADING_TIMEOUT': '1200', 'NV_LIBCUBLAS_PACKAGE': 'libcublas-11-8=11.11.3.6-1', 'TF_NUM_INTRAOP_THREADS': '1', 'NV_LIBCUBLAS_DEV_PACKAGE_NAME': 'libcublas-dev-11-8', '_': '/usr/bin/djl-serving', 'NV_CUDNN_PACKAGE_DEV': 'libcudnn8-dev=8.9.0.131-1+cuda11.8', 'NV_LIBNPP_DEV_PACKAGE': 'libnpp-dev-11-8=11.8.0.86-1', 'TRANSFORMERS_CACHE': '/tmp/.cache/huggingface/transformers', 'NV_LIBCUBLAS_DEV_PACKAGE': 'libcublas-dev-11-8=11.11.3.6-1', 'NVARCH': 'x86_64', 'DJL_CACHE_DIR': '/tmp/.djl.ai', 'MODEL_SERVER_HOME': '/opt/djl', 'NV_LIBNCCL_PACKAGE_VERSION': '2.16.2-1', 'NV_LIBNPP_PACKAGE': 'libnpp-11-8=11.8.0.86-1', 'HOSTNAME': '59b2955e7640', 'NV_CUDNN_PACKAGE': 'libcudnn8=8.9.0.131-1+cuda11.8', 'HOME': '/root', 'NV_LIBCUBLAS_VERSION': '11.11.3.6-1', 'NV_LIBNCCL_DEV_PACKAGE_VERSION': '2.16.2-1', 'LC_CTYPE': 'C.UTF-8', 'LOCAL_RANK': '0', 'RANK': '0', 'GROUP_RANK': '0', 'ROLE_RANK': '0', 'ROLE_NAME': 'default', 'LOCAL_WORLD_SIZE': '1', 'WORLD_SIZE': '1', 'GROUP_WORLD_SIZE': '1', 'ROLE_WORLD_SIZE': '1', 'MASTER_ADDR': '127.0.0.1', 'MASTER_PORT': '19000', 'TORCHELASTIC_RESTART_COUNT': '0', 'TORCHELASTIC_MAX_RESTARTS': '0', 'TORCHELASTIC_RUN_ID': 'none', 'TORCHELASTIC_USE_AGENT_STORE': 'True', 'NCCL_ASYNC_ERROR_HANDLING': '1', 'TORCHELASTIC_ERROR_FILE': '/tmp/torchelastic_3twom9hb/none_tgbdh6lu/attempt_0/0/error.json', 'CUDA_MODULE_LOADING': 'LAZY'})
[INFO ] PyProcess - W-3893-gptj-stdout: environ({'PATH': '/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin', 'NV_LIBCUSPARSE_VERSION': '11.7.5.86-1', 'PYTORCH_KERNEL_CACHE_PATH': '/tmp/.cache', 'HUGGINGFACE_HUB_CACHE': '/tmp/.cache/huggingface/hub', 'NCCL_VERSION': '2.16.2-1', 'NV_NVTX_VERSION': '11.8.86-1', 'NVIDIA_REQUIRE_CUDA': 'cuda>=11.8 brand=tesla,driver>=450,driver<451 brand=tesla,driver>=470,driver<471 brand=unknown,driver>=470,driver<471 brand=nvidia,driver>=470,driver<471 brand=nvidiartx,driver>=470,driver<471 brand=geforce,driver>=470,driver<471 brand=geforcertx,driver>=470,driver<471 brand=quadro,driver>=470,driver<471 brand=quadrortx,driver>=470,driver<471 brand=titan,driver>=470,driver<471 brand=titanrtx,driver>=470,driver<471 brand=tesla,driver>=510,driver<511 brand=unknown,driver>=510,driver<511 brand=nvidia,driver>=510,driver<511 brand=nvidiartx,driver>=510,driver<511 brand=geforce,driver>=510,driver<511 brand=geforcertx,driver>=510,driver<511 brand=quadro,driver>=510,driver<511 brand=quadrortx,driver>=510,driver<511 brand=titan,driver>=510,driver<511 brand=titanrtx,driver>=510,driver<511 brand=tesla,driver>=515,driver<516 brand=unknown,driver>=515,driver<516 brand=nvidia,driver>=515,driver<516 brand=nvidiartx,driver>=515,driver<516 brand=geforce,driver>=515,driver<516 brand=geforcertx,driver>=515,driver<516 brand=quadro,driver>=515,driver<516 brand=quadrortx,driver>=515,driver<516 brand=titan,driver>=515,driver<516 brand=titanrtx,driver>=515,driver<516', 'JAVA_OPTS': '-Xmx1g -Xms1g -XX:+ExitOnOutOfMemoryError -XX:-UseContainerSupport', 'NV_LIBCUBLAS_PACKAGE_NAME': 'libcublas-11-8', 'NVIDIA_PRODUCT_NAME': 'CUDA', 'MXNET_ENGINE_TYPE': 'NaiveEngine', 'NV_CUDNN_VERSION': '8.9.0.131', 'CUDA_VISIBLE_DEVICES': '2', 'LD_LIBRARY_PATH': '/usr/local/nvidia/lib:/usr/local/nvidia/lib64', 'NV_NVPROF_DEV_PACKAGE': 'cuda-nvprof-11-8=11.8.87-1', 'PWD': '/opt/djl', 'NV_LIBNPP_VERSION': '11.8.0.86-1', 'NV_LIBNCCL_PACKAGE': 'libnccl2=2.16.2-1+cuda11.8', 'NV_LIBNPP_DEV_VERSION': '11.8.0.86-1', 'TF_CPP_MIN_LOG_LEVEL': '1', 'PYTHONPATH': '/tmp/.djl.ai/python/0.24.0-SNAPSHOT:/opt/ml/model/gptj', 'NV_CUDA_CUDART_VERSION': '11.8.89-1', 'NV_CUDA_NSIGHT_COMPUTE_VERSION': '11.8.0-1', 'NV_LIBCUBLAS_DEV_VERSION': '11.11.3.6-1', 'OLDPWD': '/opt/djl/src/djl-serving/serving', 'NVIDIA_DRIVER_CAPABILITIES': 'compute,utility', 'OMP_NUM_THREADS': '1', 'LIBRARY_PATH': '/usr/local/cuda/lib64/stubs', 'PREDICT_TIMEOUT': '240', 'NV_CUDA_NSIGHT_COMPUTE_DEV_PACKAGE': 'cuda-nsight-compute-11-8=11.8.0-1', 'NV_CUDNN_PACKAGE_NAME': 'libcudnn8', 'LS_COLORS': 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:', 'USE_AICCL_BACKEND': 'true', 'NV_LIBNCCL_DEV_PACKAGE_NAME': 'libnccl-dev', 'SHLVL': '2', 'NV_LIBCUSPARSE_DEV_VERSION': '11.7.5.86-1', 'NV_LIBNCCL_DEV_PACKAGE': 'libnccl-dev=2.16.2-1+cuda11.8', 'NV_CUDA_LIB_VERSION': '11.8.0-1', 'JAVA_HOME': '/usr/lib/jvm/java-11-openjdk-amd64', 'NV_CUDA_CUDART_DEV_VERSION': '11.8.89-1', 'CUDA_VERSION': '11.8.0', 'NVIDIA_VISIBLE_DEVICES': 'all', 'NV_NVPROF_VERSION': '11.8.87-1', 'NV_CUDA_COMPAT_PACKAGE': 'cuda-compat-11-8', 'TERM': 'xterm', 'NV_LIBNCCL_PACKAGE_NAME': 'libnccl2', 'BITSANDBYTES_NOWELCOME': '1', 'NV_NVML_DEV_VERSION': '11.8.86-1', 'MODEL_LOADING_TIMEOUT': '1200', 'NV_LIBCUBLAS_PACKAGE': 'libcublas-11-8=11.11.3.6-1', 'TF_NUM_INTRAOP_THREADS': '1', 'NV_LIBCUBLAS_DEV_PACKAGE_NAME': 'libcublas-dev-11-8', '_': '/usr/bin/djl-serving', 'NV_CUDNN_PACKAGE_DEV': 'libcudnn8-dev=8.9.0.131-1+cuda11.8', 'NV_LIBNPP_DEV_PACKAGE': 'libnpp-dev-11-8=11.8.0.86-1', 'TRANSFORMERS_CACHE': '/tmp/.cache/huggingface/transformers', 'NV_LIBCUBLAS_DEV_PACKAGE': 'libcublas-dev-11-8=11.11.3.6-1', 'NVARCH': 'x86_64', 'DJL_CACHE_DIR': '/tmp/.djl.ai', 'MODEL_SERVER_HOME': '/opt/djl', 'NV_LIBNCCL_PACKAGE_VERSION': '2.16.2-1', 'NV_LIBNPP_PACKAGE': 'libnpp-11-8=11.8.0.86-1', 'HOSTNAME': '59b2955e7640', 'NV_CUDNN_PACKAGE': 'libcudnn8=8.9.0.131-1+cuda11.8', 'HOME': '/root', 'NV_LIBCUBLAS_VERSION': '11.11.3.6-1', 'NV_LIBNCCL_DEV_PACKAGE_VERSION': '2.16.2-1', 'LC_CTYPE': 'C.UTF-8', 'LOCAL_RANK': '0', 'RANK': '0', 'GROUP_RANK': '0', 'ROLE_RANK': '0', 'ROLE_NAME': 'default', 'LOCAL_WORLD_SIZE': '1', 'WORLD_SIZE': '1', 'GROUP_WORLD_SIZE': '1', 'ROLE_WORLD_SIZE': '1', 'MASTER_ADDR': '127.0.0.1', 'MASTER_PORT': '19003', 'TORCHELASTIC_RESTART_COUNT': '0', 'TORCHELASTIC_MAX_RESTARTS': '0', 'TORCHELASTIC_RUN_ID': 'none', 'TORCHELASTIC_USE_AGENT_STORE': 'True', 'NCCL_ASYNC_ERROR_HANDLING': '1', 'TORCHELASTIC_ERROR_FILE': '/tmp/torchelastic_bc3xj201/none_pr4il2re/attempt_0/0/error.json', 'CUDA_MODULE_LOADING': 'LAZY'})
[INFO ] PyProcess - W-3892-gptj-stdout: Python engine started.
[INFO ] PyProcess - W-3895-gptj-stdout: environ({'PATH': '/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin', 'NV_LIBCUSPARSE_VERSION': '11.7.5.86-1', 'PYTORCH_KERNEL_CACHE_PATH': '/tmp/.cache', 'HUGGINGFACE_HUB_CACHE': '/tmp/.cache/huggingface/hub', 'NCCL_VERSION': '2.16.2-1', 'NV_NVTX_VERSION': '11.8.86-1', 'NVIDIA_REQUIRE_CUDA': 'cuda>=11.8 brand=tesla,driver>=450,driver<451 brand=tesla,driver>=470,driver<471 brand=unknown,driver>=470,driver<471 brand=nvidia,driver>=470,driver<471 brand=nvidiartx,driver>=470,driver<471 brand=geforce,driver>=470,driver<471 brand=geforcertx,driver>=470,driver<471 brand=quadro,driver>=470,driver<471 brand=quadrortx,driver>=470,driver<471 brand=titan,driver>=470,driver<471 brand=titanrtx,driver>=470,driver<471 brand=tesla,driver>=510,driver<511 brand=unknown,driver>=510,driver<511 brand=nvidia,driver>=510,driver<511 brand=nvidiartx,driver>=510,driver<511 brand=geforce,driver>=510,driver<511 brand=geforcertx,driver>=510,driver<511 brand=quadro,driver>=510,driver<511 brand=quadrortx,driver>=510,driver<511 brand=titan,driver>=510,driver<511 brand=titanrtx,driver>=510,driver<511 brand=tesla,driver>=515,driver<516 brand=unknown,driver>=515,driver<516 brand=nvidia,driver>=515,driver<516 brand=nvidiartx,driver>=515,driver<516 brand=geforce,driver>=515,driver<516 brand=geforcertx,driver>=515,driver<516 brand=quadro,driver>=515,driver<516 brand=quadrortx,driver>=515,driver<516 brand=titan,driver>=515,driver<516 brand=titanrtx,driver>=515,driver<516', 'JAVA_OPTS': '-Xmx1g -Xms1g -XX:+ExitOnOutOfMemoryError -XX:-UseContainerSupport', 'NV_LIBCUBLAS_PACKAGE_NAME': 'libcublas-11-8', 'NVIDIA_PRODUCT_NAME': 'CUDA', 'MXNET_ENGINE_TYPE': 'NaiveEngine', 'NV_CUDNN_VERSION': '8.9.0.131', 'CUDA_VISIBLE_DEVICES': '2', 'LD_LIBRARY_PATH': '/usr/local/nvidia/lib:/usr/local/nvidia/lib64', 'NV_NVPROF_DEV_PACKAGE': 'cuda-nvprof-11-8=11.8.87-1', 'PWD': '/opt/djl', 'NV_LIBNPP_VERSION': '11.8.0.86-1', 'NV_LIBNCCL_PACKAGE': 'libnccl2=2.16.2-1+cuda11.8', 'NV_LIBNPP_DEV_VERSION': '11.8.0.86-1', 'TF_CPP_MIN_LOG_LEVEL': '1', 'PYTHONPATH': '/tmp/.djl.ai/python/0.24.0-SNAPSHOT:/opt/ml/model/gptj', 'NV_CUDA_CUDART_VERSION': '11.8.89-1', 'NV_CUDA_NSIGHT_COMPUTE_VERSION': '11.8.0-1', 'NV_LIBCUBLAS_DEV_VERSION': '11.11.3.6-1', 'OLDPWD': '/opt/djl/src/djl-serving/serving', 'NVIDIA_DRIVER_CAPABILITIES': 'compute,utility', 'OMP_NUM_THREADS': '1', 'LIBRARY_PATH': '/usr/local/cuda/lib64/stubs', 'PREDICT_TIMEOUT': '240', 'NV_CUDA_NSIGHT_COMPUTE_DEV_PACKAGE': 'cuda-nsight-compute-11-8=11.8.0-1', 'NV_CUDNN_PACKAGE_NAME': 'libcudnn8', 'LS_COLORS': 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:', 'USE_AICCL_BACKEND': 'true', 'NV_LIBNCCL_DEV_PACKAGE_NAME': 'libnccl-dev', 'SHLVL': '2', 'NV_LIBCUSPARSE_DEV_VERSION': '11.7.5.86-1', 'NV_LIBNCCL_DEV_PACKAGE': 'libnccl-dev=2.16.2-1+cuda11.8', 'NV_CUDA_LIB_VERSION': '11.8.0-1', 'JAVA_HOME': '/usr/lib/jvm/java-11-openjdk-amd64', 'NV_CUDA_CUDART_DEV_VERSION': '11.8.89-1', 'CUDA_VERSION': '11.8.0', 'NVIDIA_VISIBLE_DEVICES': 'all', 'NV_NVPROF_VERSION': '11.8.87-1', 'NV_CUDA_COMPAT_PACKAGE': 'cuda-compat-11-8', 'TERM': 'xterm', 'NV_LIBNCCL_PACKAGE_NAME': 'libnccl2', 'BITSANDBYTES_NOWELCOME': '1', 'NV_NVML_DEV_VERSION': '11.8.86-1', 'MODEL_LOADING_TIMEOUT': '1200', 'NV_LIBCUBLAS_PACKAGE': 'libcublas-11-8=11.11.3.6-1', 'TF_NUM_INTRAOP_THREADS': '1', 'NV_LIBCUBLAS_DEV_PACKAGE_NAME': 'libcublas-dev-11-8', '_': '/usr/bin/djl-serving', 'NV_CUDNN_PACKAGE_DEV': 'libcudnn8-dev=8.9.0.131-1+cuda11.8', 'NV_LIBNPP_DEV_PACKAGE': 'libnpp-dev-11-8=11.8.0.86-1', 'TRANSFORMERS_CACHE': '/tmp/.cache/huggingface/transformers', 'NV_LIBCUBLAS_DEV_PACKAGE': 'libcublas-dev-11-8=11.11.3.6-1', 'NVARCH': 'x86_64', 'DJL_CACHE_DIR': '/tmp/.djl.ai', 'MODEL_SERVER_HOME': '/opt/djl', 'NV_LIBNCCL_PACKAGE_VERSION': '2.16.2-1', 'NV_LIBNPP_PACKAGE': 'libnpp-11-8=11.8.0.86-1', 'HOSTNAME': '59b2955e7640', 'NV_CUDNN_PACKAGE': 'libcudnn8=8.9.0.131-1+cuda11.8', 'HOME': '/root', 'NV_LIBCUBLAS_VERSION': '11.11.3.6-1', 'NV_LIBNCCL_DEV_PACKAGE_VERSION': '2.16.2-1', 'LC_CTYPE': 'C.UTF-8', 'LOCAL_RANK': '0', 'RANK': '0', 'GROUP_RANK': '0', 'ROLE_RANK': '0', 'ROLE_NAME': 'default', 'LOCAL_WORLD_SIZE': '1', 'WORLD_SIZE': '1', 'GROUP_WORLD_SIZE': '1', 'ROLE_WORLD_SIZE': '1', 'MASTER_ADDR': '127.0.0.1', 'MASTER_PORT': '19002', 'TORCHELASTIC_RESTART_COUNT': '0', 'TORCHELASTIC_MAX_RESTARTS': '0', 'TORCHELASTIC_RUN_ID': 'none', 'TORCHELASTIC_USE_AGENT_STORE': 'True', 'NCCL_ASYNC_ERROR_HANDLING': '1', 'TORCHELASTIC_ERROR_FILE': '/tmp/torchelastic_th2h_egj/none_hveusn2j/attempt_0/0/error.json', 'CUDA_MODULE_LOADING': 'LAZY'})
[INFO ] PyProcess - W-3893-gptj-stdout: Python engine started.
[INFO ] PyProcess - W-3895-gptj-stdout: Python engine started.
```